packages/kbn-user-profile-components/src/services.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, useContext } from 'react';
+import { NotificationsStart, ToastOptions } from '@kbn/core-notifications-browser';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+import type { I18nStart } from '@kbn/core-i18n-browser';
+
+import { ThemeServiceStart } from '@kbn/react-kibana-context-common';
+import type { UserProfileAPIClient } from './types';
+
+type NotifyFn = (
+  data: { title: string; text?: JSX.Element },
+  options?: { durationMs?: number }
+) => void;
+
+export interface Services {
+  userProfileApiClient: UserProfileAPIClient;
+  notifySuccess: NotifyFn;
+}
+
+const UserProfilesContext = React.createContext<Services | null>(null);
+
+/**
+ * Abstract external service Provider.
+ */
+export const UserProfilesProvider: FC<Services> = ({ children, ...services }) => {
+  return <UserProfilesContext.Provider value={services}>{children}</UserProfilesContext.Provider>;
+};
+
+/**
+ * Kibana-specific service types.
+ */
+export interface UserProfilesKibanaDependencies {
+  /** CoreStart contract */
+  core: {
+    notifications: NotificationsStart;
+    i18n: I18nStart;
+    theme: ThemeServiceStart;
+  };
+  security: {
+    userProfiles: UserProfileAPIClient;
+  };
+}
+
+/**
+ * Kibana-specific Provider that maps to known dependency types.
+ */
+export const UserProfilesKibanaProvider: FC<UserProfilesKibanaDependencies> = ({
```

Comment:
Is there a reason we need to create a new context provider for this functionality? 

You should be able to read `http` from the existing Kibana context provider and then instantiate your own `UserProfilesApiClient` directly in the `useUpdateUserProfile` hook. 

Same for the `onSuccess` callback.

Reply:
See my comment above about providing stateful Kibana dependencies to packages.